R/join_ald_scenario.R

```diff
@@ -48,14 +48,93 @@ join_ald_scenario <- function(data,
   ald <- modify_at_(ald, "technology", tolower)
   out <- data %>%
     left_join(ald, by = ald_columns()) %>%
-    inner_join(scenario, by = scenario_columns()) %>%
-    warn_if_has_zero_rows("Joining `scenario` outputs 0 rows.") %>%
-    mutate(plant_location = tolower(.data$plant_location)) %>%
-    inner_join(
-      region_isos,
-      by = c("region", "plant_location" = "isos", "scenario_source")
-    ) %>%
+    mutate(plant_location = tolower(plant_location)) %>%
+    left_join(region_isos, by = c("plant_location" = "isos")) %>%
     warn_if_has_zero_rows("Joining `region_isos` outputs 0 rows.")
+  all_company <- unique(out$name_ald)
+  all_region <- unique(out$region)
+  for (one_region in all_region) {
+    # print(one_region)
+    for (company in all_company) {
+      filtered_data <- out %>%
+        filter(
+          name_ald == company
+        )
+      sector_company <- filtered_data$sector_ald[1]
+      all_tech <- green_or_brown %>%
+        filter(sector == sector_company)
+      all_tech <- all_tech$technology
+      green_tech <- green_or_brown %>%
+        filter(
+          sector == sector_company,
+          green_or_brown == "green"
+        )
+      green_tech <- green_tech$technology
+      if (sector_company == "automotive") {
```

Comment:
Line 73-91: I'm not sure those line are useful as in the next data release, differenciation is done at sector level (i.e. `sector` is automotive or hdv) and no more at technology level (i.e.` sector` is automotive and `technology` is electric or electric_hdv)

Reply:
@Antoine-Lalechere can you add a comment here with the URL to the issue or PR that points to the changes you refer to as the "next data release"? 

Whatever we do here needs to be backward compatible -- i.e. support the new AND old version of the data you refer here.

Reply:
Something of this type: https://github.com/2DegreesInvesting/r2dii.data/pull/237
It's necessary with the oldest format and useless with new format so if it needs to be backward compatible then we should keep it